src/containers/home/home.component.jsx

```diff
@@ -1,166 +1,59 @@
-import React, { useEffect, useState } from "react";
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, { useState, useEffect } from "react";
 import { useSelector } from "react-redux";
 import Post from "containers/post";
 import Profile from "containers/profile";
 import Modal from "components/modal";
 import Navbar from "containers/navbar";
 
 const Home = ({ fetchPostsThunk, fetchUserAvatarThunk, view }) => {
-  const [displayPosts, setDisplayPosts] = useState([]);
-  const [selectDisplayPostOption, setSelectDisplayPostOption] = useState(
-    "Newest"
-  );
-  const [viewByTimeframeTime, setViewByTimeframeTime] = useState(Date.now());
-  const [viewByTimeframe, setViewByTimeframe] = useState("All-time");
+  // Check whether user has a username
   const userFromState = useSelector((state) => state.users.user);
   const doesUserHaveUsername = !!userFromState.username;
-  const postsFromState = useSelector((state) => state.posts);
 
-  const handleDisplayPostChange = () => {
-    const viewOption = document.getElementById("select-view-option").value;
-    setSelectDisplayPostOption(viewOption);
-    const timeframe = document.getElementById("select-view-option-timeline")
-      .value;
-    setViewByTimeframe(timeframe);
-    if (timeframe === "All-time") {
-      setViewByTimeframeTime(Date.now());
-    } else if (timeframe === "Today") {
-      setViewByTimeframeTime(86400000);
-    } else if (timeframe === "This week") {
-      setViewByTimeframeTime(604800000);
-    }
-  };
+  // Information about the posts
+  const postsFromState = useSelector((state) => state.posts.posts); //Fix this posts.posts shit
```

Comment:
I think I know how to fix this in redux. I'll stab at it

Reply:
AWesome